colossalai/zero/gemini/gemini_ddp.py

```diff
@@ -430,7 +444,18 @@ def _get_chunk_to_save_data(self, chunk: Chunk, only_rank_0: bool) -> Dict:
             record_tensor = torch.empty([0])
             record_flag = (not only_rank_0) | (dist.get_rank(chunk.torch_pg) == 0)
             if record_flag:
-                record_tensor = temp_chunk[tensor_info.offset : tensor_info.end].view(tensor.shape).cpu()
+                record_tensor = temp_chunk[tensor_info.offset : tensor_info.end].view(tensor.shape).cuda()
+                if is_distributed_tensor(tensor):
+                    global_shape = get_global_shape(tensor)
+                    device_mesh = get_device_mesh(tensor)
+                    shard_spec = get_sharding_spec(tensor)
+                    record_tensor = init_as_dtensor(record_tensor, 
+                                                      device_mesh=device_mesh, 
+                                                      sharding_spec=shard_spec,
+                                                      global_shape = global_shape)
+                elif is_customized_distributed_tensor(tensor):
+                    init_tensor_as_customization_distributed(record_tensor, shard_fn=tensor.shard_fn, gather_fn=tensor.gather_fn)
+                record_tensor = gather_distributed_param(record_tensor, keep_vars=False)           
 
             assert tensor not in chunk_to_save_data
             chunk_to_save_data[tensor] = record_tensor
```

Comment:
`record_tensor` needs to be put into cpu

Reply:
updated.